File: 144-binary-tree-preorder-traversal/144-binary-tree-preorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        Stack<TreeNode> nodes = new Stack<>();
        List<Integer> returnVal = new ArrayList<>();
        
        if (root == null) {
            return returnVal;
        }
        
        nodes.push(root);
        
        while(!nodes.isEmpty()) {
            TreeNode curr = nodes.pop();
            returnVal.add(curr.val);
            
            if(curr.right != null) {
                nodes.push(curr.right);
            }
            
            if(curr.left != null) {
                nodes.push(curr.left);
            }
            
           
        }
        
        return returnVal;
    }
}

File: 144-binary-tree-preorder-traversal/144-binary-tree-preorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        
        lst = []
        stack = []
        if root is None:
            return lst
        stack.append(root)
        
        while(stack):
            s = stack.pop()
            
            lst.append(s.val)
            if(s.right is not None):
                stack.append(s.right)
            if(s.left is not None):
                stack.append(s.left)
        
        return lst